app/services/field_parser.rb

```diff
@@ -0,0 +1,16 @@
+module FieldParser
```

Comment:
why is a module the right vehicle for this behaviour?

Reply:
It's interesting. It's code extracted from a Module, into a Module, included in a controller, a model concern, and a library.

It sounds like a utility helper. For a utility helper a lib, or service can be an okay place for it but then it shouldn't be included like a module it should be accessed as a class method or a field parsing object than knows how to return parsed data.

The second option a `parsing object` would be more conducive to the original implementation. The `HasFields` concern was included on objects like Notes, Cards etc. objects that have fields that need parsing. We've now got data coming in to a controller that needs fields parsed. OOP would say we pass the data into an object that also `HasFields` and can return the parsed data for us, instead of trying to apply the function to different code entry points.

Reply:
> OOP would say we pass the data into an object that also HasFields and can return the parsed data for us, instead of trying to apply the function to different code entry points.

This is the current implementation in the controller: `Note.parse_fields(params[:source])`

But @etdsoft mentioned in the task:

> I’m not a fan of the current implementation of calling Note.parse_fields to return a Hash though (as it has nothing to do with Note). I think we’d be better of with HasFields::ClassMethods::parse_fields but it seems like a smell anyway.

I'm thinking the best middleground here is to define a _class_ method for the `HasFields` module:

```
module HasFields
  def self.parse_fields
   ..
```

and use it as `HasFields.parse_fields`, which sounds like your first option.

Reply:
For the sake of conversation:

> as it has nothing to do with Note

I'd agree with that. Using a note to do the work is a misuse of `Note`.
Creating an intermediary object that also `HasFields` for this purpose is what I was thinking. 

```
class EditorData
  include HasFields

  def initialize(data)
    @data = data
  end

  def fields
    parse_fields(@data)
  end
end

Now we have an object that can act like the other `HasFields` objects, but stands alone in purpose, and description.

Additionally, I'm not opposed to `parse_fields` becoming a class method. The fact it only operates on non-internal data means it was never really an object method in the firstplace.

Reply:
fwiw, I was thinking of a singleton class with a class method. You wouldn't be able to mix it in from other classes, but you'd be able to call the singleton anyway.